VogtObserver/Generator.cs

```diff
@@ -4,7 +4,34 @@
 
 namespace VogtObserver
 {
-    class Generator
+    class Generator : IObservable
     {
+
+          private List<IObserver> _observers;
+
+        public void Add(IObserver o)
+        {
+            _observers.Add(o);
+        }
+
+        public void Notify()
+        {
+            foreach (IObserver o in _observers)
+            {
+                o.Update();
+            }
+        }
+
+        public void Remove(IObserver o)
+        {
+            _observers.Remove(o);
+        }
+
+        public int GetRandomInt()
+        {
+            var random = new Random();
+
+            return random.Next(1, 1000);
```

Comment:
I'm pretty sure random should be instantiated once per application.

What I do in mine is put this in the constructor:
```
class Generator {
  private readonly Random _random;
  public Generator() {
    _random = new Random();
  }
}
```

Then you can access the same object w/o having to create a new one every single time (**that wastes memory**!)